app/views/contexts/section_summary_context.py

```diff
@@ -95,6 +95,15 @@ def _list_summary_element(self, summary, current_location, section) -> Mapping:
             section, for_list=summary["for_list"]
         )[0]
 
+        primary_person_blocks = self._schema.get_list_collectors_for_list(
+            section, for_list=summary["for_list"], primary=True
+        )
+        primary_person_edit_block_id = (
+            primary_person_blocks[0]["add_or_edit_block"]["id"]
+            if len(primary_person_blocks) > 0
+            else None
+        )
```

Comment:
We could save fetching the primary person list collector block by checking if the list store for the `for_list` contains a primary person. I think we should also use a generator. 

To add to that, if we check the list count, then we can use the normal list collector when we know it is available, i.e. when count is above 1, therefore a better UX given as it changes the text for the user.
```py
        primary_person_edit_block_id = None
        current_list = self._list_store[summary["for_list"]]
        if len(current_list) == 1 and current_list.primary_person:
            primary_person_edit_block_id = self._schema.get_list_collectors_for_list(
                section, for_list=summary["for_list"], primary=True
            )[0]["add_or_edit_block"]["id"]
```